src/trace/listener.ts

```diff
@@ -11,6 +11,8 @@ import { Source, ddtraceVersion } from "./constants";
 import { patchConsole } from "./patch-console";
 import { SpanContext, TraceOptions, TracerWrapper } from "./tracer-wrapper";
 
+export type TraceExtractor = (event: any, context: Context) => TraceContext | undefined;
```

Comment:
I don't think we need `| undefined` at the end here. Having it be optional in trace config is enough.